### DISCUSSION TITLE
Error running make on MUMPS when compiling petsc

### DISCUSSION THREAD

**Discussion by tengzhang48** [idaholab/moose#20524] [Q&A General] [answered]
Hello,

I am trying to update the Moose on my account on XSEDE Expanse cluster but have error running make on MUMPS when compiling petsc. I previously successfully install it on the cluster. The error information is

"dlr_stats.F:343:12:

  343 |         use DMUMPS_STRUC_DEF
      |            1
Fatal Error: Cannot open module file 'dmumps_struc_def.mod' for reading at (1): No such file or directory
compilation terminated."

But I do found "dmumps_struc_def.mod" file in the moose/petsc/arch-moose/externalpackages/MUMPS_5.3.5/src. Also, I just followed the instruction on the moose website for the installations and did not change anything. 

Could you please help me take a look at this?

Thanks a lot.

Best,
Teng

**Comment by GiudGiud** [comment, marked as answer]
Hello

Sorry for the delay. 
Could you please try again now? The petsc module has been updated.
The installation instructions for clusters: 
https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html

Guillaume

**Reply by tengzhang48** [reply]
I tried to re-compile the software. Now petsc works. But I have the following error when installing libmesh 
" /petscsys.h:535:25: error: invalid conversion from 'MPI_Comm' {aka 'ompi_communicator_t*'} to 'TIMPI::communicator' {aka 'int'} [-fpermissive]"
Could you please help me check this? Thanks a lot.

P.S. I previously successfully installed moose by disabling mumps. But I covered my previous installation with the unsuccessful one now.

**Reply by GiudGiud** [reply]
Weird. What mpi distribution are you using?
What are the outputs of 
```
mpicc --version
echo $CC $CXX $FC
```

**Reply by tengzhang48** [reply]
I used openmpi yesterday. I tried intel-mpi today, and it works. Most tests can be passed. I received 14 errors for the last tests. I guess they are more advanced functions and probably I will not use them in the near future. If they can be easily fixed, it will be great to hear your suggestions on this. If not, you can close this issue. I will post related questions later if I indeed use these functions. 

Thanks a lot.

Below are the failed test:


"mesh/split_uniform_refine.square ...................................... [FINISHED,min_cpus=3] FAILED (TIMEOUT)
partitioners/petsc_partitioner.ptscotch_weight_both ................... [FINISHED,min_cpus=4] FAILED (TIMEOUT)
partitioners/petsc_partitioner.parmetis_weight_side ................... [FINISHED,min_cpus=4] FAILED (TIMEOUT)
partitioners/petsc_partitioner.parmetis_weight_both ................... [FINISHED,min_cpus=4] FAILED (TIMEOUT)
meshgenerators/distributed_rectilinear/dmg_displaced_mesh.pbc/adaptivity_nemesis  [min_cpus=4] FAILED (ERRMSG)
partitioners/petsc_partitioner.ptscotch ......................................... [min_cpus=4] FAILED (ERRMSG)
mesh/mesh_only.mesh_only_checkpoint ............................................. [min_cpus=3] FAILED (ERRMSG)
postprocessors/num_elems.test_split ............................................. [min_cpus=4] FAILED (ERRMSG)
postprocessors/num_nodes.test_split ............................................. [min_cpus=4] FAILED (ERRMSG)
partitioners/petsc_partitioner.ptscotch_weight_elment ........................... [min_cpus=4] FAILED (ERRMSG)
mesh/nemesis.nemesis_repartitioning_test ........................................ [min_cpus=4] FAILED (ERRMSG)
partitioners/petsc_partitioner.ptscotch_weight_side ............................. [min_cpus=4] FAILED (ERRMSG)
partitioners/petsc_partitioner.parmetis ......................................... [min_cpus=4] FAILED (ERRMSG)
partitioners/petsc_partitioner.parmetis_weight_element .......................... [min_cpus=4] FAILED (ERRMSG)"

**Reply by GiudGiud** [reply]
It looks like the petsc partitioners are failing which is not a good sign. 

If you could please paste the actual error messages we could see what that’s about. 
to rerun only those tests you can do
‘./run_tests —failed-tests’

Officially we don’t support intel compilers btw. I m not sure why they are working for you.

**Reply by tengzhang48** [reply]
I re-downloaded moose and compiled again with openmpi. Now it works and all the tests can pass. It is strange that I did not change anything this time. I realized I should use the job script to run the tests. Yesterday, I used interactive job, which is based on "srun". Anyway, you mentioned that intel-mpi may not be a good option (I actually used gcc + intel-mpi, which may be a wried combination). It is good that now the default installation works.

Thanks for your help.
